### PR TITLE
Limit auto-import to recent versions

### PR DESCRIPTION
**Pull request by neilcsmith-net**
Address #6135 by limiting auto-upgrade importing of settings to latest 3 versions.  Does not stop manual importing of settings from earlier versions.

Have tested this ignores 8.2 and 13, but suggests more recent.

### CODE REVIEW

File: nb/o.n.upgrader/src/org/netbeans/upgrade/AutoUpgrade.java [hunk 1]
```diff
@@ -65,10 +63,6 @@ public static void main (String[] args) throws Exception {
         
         // try new place
         File sourceFolder = checkPreviousOnOsSpecificPlace (NEWER_VERSION_TO_CHECK);
-        if (sourceFolder == null) {
-            // try former place
-            sourceFolder = checkPrevious (VERSION_TO_CHECK);
```
Comment (mbien) [inline nb/o.n.upgrader/src/org/netbeans/upgrade/AutoUpgrade.java:70]: i think `checkPrevious` is now no longer used and can be removed, same with `upgradeBuildProperties`, `doNonStandardUpgrade`, potentially more

File: nb/o.n.upgrader/src/org/netbeans/upgrade/AutoUpgrade.java [hunk 2]
```diff
@@ -91,21 +85,11 @@ private static void upgradeBuildProperties(final File sourceFolder, final String
...
-    static final List<String> PRE_APACHE_NEWER_VERSION_TO_CHECK =
-            Arrays.asList ("8.2", "8.1", "8.0.2", "8.0.1", "8.0", "7.4", "7.3.1", "7.3", "7.2.1", "7.2"); //NOI18N
-     // XXX: copy to autoupgrade.pluginimporter
-    
     static final Comparator<String> APACHE_VERSION_COMPARATOR = (v1, v2) -> new SpecificationVersion(v1).compareTo(new SpecificationVersion(v2));
     
     static final List<String> APACHE_VERSION_TO_CHECK = Arrays.asList(NbBundle.getMessage(AutoUpgrade.class, "apachenetbeanspreviousversion").split(",")).stream().sorted(APACHE_VERSION_COMPARATOR.reversed()).collect(Collectors.toList());
     
-    static final List<String> NEWER_VERSION_TO_CHECK = Stream.concat(APACHE_VERSION_TO_CHECK.stream(), PRE_APACHE_NEWER_VERSION_TO_CHECK.stream()).collect(Collectors.toList());
+    static final List<String> NEWER_VERSION_TO_CHECK = APACHE_VERSION_TO_CHECK.stream().collect(Collectors.toList());
```
Comment (mbien) [inline nb/o.n.upgrader/src/org/netbeans/upgrade/AutoUpgrade.java:92]: `APACHE_VERSION_TO_CHECK` can be used now instead, `NEWER_VERSION_TO_CHECK` is just a copy at this point

**Review by mbien — APPROVED** [review]
good change! left some cleanup comments inline.

File: nb/o.n.upgrader/src/org/netbeans/upgrade/AutoUpgrade.java (hunk 1 above)
Comment (neilcsmith-net) [reply nb/o.n.upgrader/src/org/netbeans/upgrade/AutoUpgrade.java:70]: True!  Done heavier cleanup, and resolved a couple of other issues that were hinted - try-with-resources (missing close on exception) and new Integer.

File: nb/o.n.upgrader/src/org/netbeans/upgrade/AutoUpgrade.java (hunk 2 above)
Comment (neilcsmith-net) [reply nb/o.n.upgrader/src/org/netbeans/upgrade/AutoUpgrade.java]: Doh! :laughing:  Not quite sure how that passed me by at the time ...

**Review by mbien — APPROVED** [review]
nice cleanup!

**Review by ebarboni — APPROVED** [review]
LGTM, maybe we should also do something on the "web" to avoir getting all the version there :p

**Comment by neilcsmith-net** [comment]
Thanks @ebarboni On web better if we could static archive and permalink the old versions, as well as limit the number of previous versions shown in the top bar.

**Pull request merged**